src/com/projectkorra/projectkorra/firebending/FireBlast.java

```diff
@@ -161,11 +165,15 @@ public boolean checkLocation(Block block) {
 				final BlastFurnace blastF = (BlastFurnace) block.getState();
 				blastF.setBurnTime((short) 800);
 				blastF.update();
-			} else if (block instanceof Campfire) {
-				final Campfire campfire = (Campfire) block.getBlockData();
-				if(!campfire.isLit()) {
-					if(block.getType() != Material.SOUL_CAMPFIRE || bPlayer.canUseSubElement(SubElement.BLUE_FIRE)) {
-						campfire.setLit(true);
+			} else if (block.getBlockData() instanceof Lightable) {
+				final Lightable lightable = (Lightable) block.getBlockData();
+				if (!lightable.isLit()) {
+					if (block.getType() != Material.SOUL_CAMPFIRE || bPlayer.canUseSubElement(SubElement.BLUE_FIRE)) {
+						lightable.setLit(true);
+						block.setBlockData(lightable);
+					} else if (block.getType().name().toLowerCase().endsWith("candle") && lightCandles) {

```

Comment:
Does this if statement even trigger? Looks to me like it won't since the if statement above is enough

Reply:
it does https://youtu.be/gwK5GRqVm7c